osf/management/commands/import_EGAP.py

```diff
@@ -58,21 +59,28 @@ def get_creator_auth_header(creator_username):
     return creator, {'Authorization': 'Bearer {}'.format(token.token_id)}
 
 
-def create_node_from_project_json(egap_assets_path, epag_project_dir, creator):
-    with open(os.path.join(egap_assets_path, epag_project_dir, 'project.json'), 'r') as fp:
+def create_node_from_project_json(egap_assets_path, egap_project_dir, creator):
+    with open(os.path.join(egap_assets_path, egap_project_dir, 'project.json'), 'r') as fp:
         project_data = json.load(fp)
         title = project_data['title']
         node = Node(title=title, creator=creator)
         node.save()  # must save before adding contribs for auth reasons
 
         for contributor in project_data['contributors']:
-            node.add_contributor_registered_or_not(
-                Auth(creator),
-                full_name=contributor['name'],
-                email=contributor['email'],
-                permissions=WRITE,
-                send_email='false'
-            )
+            try:
+                node.add_contributor_registered_or_not(
+                    Auth(creator),
+                    full_name=contributor['name'],
+                    email=contributor['email'].strip(),
+                    permissions=WRITE,
+                    send_email='false'
+                )
+            except Exception as err:
+                logger.error(
+                    'There was an error adding a contributor with the following email: {}'.format(contributor['email'])
+                )
+                logger.info(str(err))
+                continue
```

Comment:
Will this continue creating the project etc without the contributor? Ideally, it should skip this project entirely with the specific error that it was user creation that broke it. Would re-raising the error be a good idea here, so it can be caught by the higher `try` block?

Reply:
The project will have been initiated and saved here (https://github.com/CenterForOpenScience/osf.io/pull/9207/files/fbb12607b37501e3e899cd93ca13112eeb39035d#diff-398515fad586d01fcf62caffc758e479R66-R67). Yes, re-raising the error could work well here. I'll look into it

Reply:
This has been addressed in the most recent commit